### PR TITLE
clients/java: describe job worker polling and backoff

### LINKED ISSUE #183: [java client] add basic documentation about job worker backoff
In https://github.com/zeebe-io/zeebe/pull/6093 backoff mechanism was introduced to the job worker in the java client. Users should be aware that this functionality exists, and that they can configure it. 

Configuration details are already described in the JavaDoc, to assist the user when actually configuring it, but there is nothing pointing the user to look for this specific method (i.e. `JobWorkerBuilderStep3.backoffSupplier()`).

The documentation should describe the basic functionality and possibility for configuration.

### PR DESCRIPTION
**Pull request by korthout**
Adds a description of the java client's job worker polling behaviour and how to configure its backoff strategy.

closes #183

### CODE REVIEW

File: docs/product-manuals/clients/java-client/job-worker.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+## Related Resources
+
+- [Job worker basics](/product-manuals/concepts/job-workers.md)
+
+## The java client's job worker
+
+The java client provides a job worker that takes care of polling for available jobs and allows users to focus on writing code to handle the activated jobs.
+
+Generally, the job worker:
+- polls for jobs before it runs out
```
Comment (npepinpe) [inline docs/product-manuals/clients/java-client/job-worker.md:15]: Before it runs out...of jobs? ...on its family after going for a pack of cigarette at the corner store? :thinking:

File: docs/product-manuals/clients/java-client/job-worker.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+- waits for `pollInterval` milliseconds (and polls again) when there are no jobs available
+- applies a backoff strategy on polling errors
+
+## Example usage
+
+- [Open a job worker](../java-client-examples/job-worker-open.md)
+
+## Backoff configuration
+
+When a poll fails with an error response: the job worker waits for some time, afterwhich it polls again for available jobs.
```
Comment (npepinpe) [inline docs/product-manuals/clients/java-client/job-worker.md:25]:
```suggestion
When a poll fails with an error response: the job worker waits for some time, after which it polls again for available jobs.
```

**Review by npepinpe — CHANGES_REQUESTED** [review]
I would expand a bit on how the job worker polling mechanism work. I don't think it's clear from a user point of view.

- Does it start polling immediately on creation?
- When does it poll again? How often does it poll? What's the impact on my cluster and why is backoff necessary?
- How are jobs processed? What kind of executor would I want to use? (I realize this is a little out of scope though, so feel free to create a follow up issue)

**Comment by korthout** [comment]
@npepinpe Sorry for the late reply, but I've applied your review hints. I've also tried to use the style recommended by the [technical writing cheatsheet](https://github.com/camunda-cloud/camunda-cloud-documentation/blob/master/howtos/technical-writing-cheatsheet.md). Please have a look

**Comment by akeller** [comment]
Thank you for using the style guide cheatsheet! I can review for style 🙂

File: docs/product-manuals/clients/java-client/get-started.md [hunk 3]
```diff
@@ -385,3 +385,4 @@ From here there are several steps to take, depending on your preference:
   - [Handle variables as POJO](../java-client-examples/data-pojo.md)
 - Learn how to [write tests](testing.md)
 - Learn more about [BPMN workflows](/reference/bpmn-workflows/bpmn-primer.md) in general
+- Learn more about the java client's [job worker](job-worker.md)
```
Comment (akeller) [inline docs/product-manuals/clients/java-client/get-started.md:388]:
```suggestion
- Learn more about the Java client's [job worker](job-worker.md)
```

File: docs/product-manuals/clients/java-client/index.md [hunk 4]
```diff
@@ -85,6 +85,7 @@ ZeebeClient client =
 ## Next Steps
 
 - [Getting started guide](get-started.md) - Comprehensive tutorial that covers Zeebe Modeler, Operate and the Java client.
+- [Job worker](job-worker.md) - Introduction to the java client's job worker
```
Comment (akeller) [inline docs/product-manuals/clients/java-client/index.md:88]:
```suggestion
- [Job worker](job-worker.md) - Introduction to the Java client's job worker
```

File: docs/product-manuals/clients/java-client/index.md [hunk 5]
```diff
@@ -85,6 +85,7 @@ ZeebeClient client =
...
 - [Logging](logging.md) - Introduction on how to configure logging for Zeebe Client
```
Comment (akeller) [inline docs/product-manuals/clients/java-client/index.md:89]:
```suggestion
- [Logging](logging.md) - Introduction on how to configure logging for Zeebe client
```

File: docs/product-manuals/clients/java-client/job-worker.md [hunk 6]
```diff
@@ -0,0 +1,42 @@
+---
+id: job-worker
+title: "Job worker"
+---
+
+## Related Resources
```
Comment (akeller) [inline docs/product-manuals/clients/java-client/job-worker.md:6]:
```suggestion
## Related resources
```

File: docs/product-manuals/clients/java-client/job-worker.md [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+
+- [Job worker basics](/product-manuals/concepts/job-workers.md)
+
+## The java client's job worker
```
Comment (akeller) [inline docs/product-manuals/clients/java-client/job-worker.md:10]:
```suggestion
## The Java client's job worker

The Java client provides a job worker that takes care of polling for available jobs and allows you to focus on writing code to handle the activated jobs.
```

**Review by akeller — CHANGES_REQUESTED** [review]
Just a couple capitalization items from a style guide perspective.

**Review by npepinpe — APPROVED** [review]
Looks good from my side :+1:

**Comment by korthout** [comment]
@akeller thanks for the review, I've made the requested changes. Please have another look.

PS: These changes are part of the upcoming zeebe release. How do we make sure this documentation is not published before the release? Should we keep the PR from being merged, do we have some form of marking the changes as related to a next release or do we just wait with deploying the next version of the documentation until the next release is ready?

**Comment by akeller** [comment]
We have a branch for `release_0_27` that we've been using for the next Zeebe release, but I mistakenly thought it was going out this month so we have a bit of a mess PR wise right now 🙂 

What release is this for @korthout?

**Comment by korthout** [comment]
@akeller the next release is 1.0.0 (there is no 0.27 planned afaik), but I can rebase this on that branch if that's the correct way :) Please let me know what you'd like me to do

**Comment by akeller** [comment]
> @akeller the next release is 1.0.0 (there is no 0.27 planned afaik), but I can rebase this on that branch if that's the correct way (that's no effort at all) :) Please let me know what you'd like me to do

Do that please. Thank you so much! Hopefully the following release is a little more streamlined 😅

**Review by akeller — COMMENTED** [review]
Looks good from a style guide perspective!

Note - this is just a comment and not an approval in case the approval triggers an auto-merge (which we don't want).

**Comment by korthout** [comment]
@akeller I've rebased and changed the target for this PR (now pointing to `release_0_27`). Please approve and merge :)

**Review by akeller — APPROVED** [review]

**Comment by akeller** [comment]
Happy weekend @korthout!

**Pull request merged**